src/main/java/com/ericsson/ei/notifications/HttpRequest.java

```diff
@@ -81,10 +81,13 @@ HttpRequest createHttpRequest() {
     @Getter
     private HttpHeaders headers;
 
+    @Setter
+    private Encryptor encryptor;
```

Comment:
Is the setter needed for mocking? else autowire it here.
Use manual wiering (Trix for non @Components) 

private Encryptor encryptor = SpringContext.getBean(Encryptor.class);

Reply:
This can't be Autowired. I already tried it. The class itself is not a Component and wouldn't Autowire the Encryptor. I think when I made it a Component other problems followed.

Reply:
All those classes:
    private HttpRequestSender httpRequestSender
    private JenkinsCrumb jenkinsCrumb
    private UrlParser urlParser

Are autowired in this class, would be strange if this does not work with Encryptor.

Reply:
They are not autowired. What they seem to do is getting a specific bean from Spring though. I didn't know you could do this. Then I can do the same for Encryptor.

Reply:
@Autowired just gets the last created bean or am I wrong?

Reply:
Yes I suppose but you cannot use the annotation in that class. @Value is not possible to do either. The workaround here is to simply ask spring for the bean which seems to work just as well.